src/target/adiv5.c

```diff
@@ -463,39 +465,43 @@ static void adiv5_component_probe(ADIv5_AP_t *ap, uint32_t addr, int recursion,
 
 	/* CIDR preamble sanity check */
 	if ((cidr & ~CID_CLASS_MASK) != CID_PREAMBLE) {
-		DEBUG_WARN("%s%d 0x%08" PRIx32": 0x%08" PRIx32
-			  " <- does not match preamble (0x%X)\n",
-			  indent + 1, num_entry, addr, cidr, CID_PREAMBLE);
+		DEBUG_WARN("%s%d 0x%08" PRIx32 ": 0x%08" PRIx32 " <- does not match preamble (0x%X)\n", indent + 1, num_entry,
+			addr, cidr, CID_PREAMBLE);
 		return;
 	}
 
-	uint64_t pidr = adiv5_ap_read_pidr(ap, addr);
 	/* Extract Component ID class nibble */
-	uint32_t cid_class = (cidr & CID_CLASS_MASK) >> CID_CLASS_SHIFT;
+	const uint32_t cid_class = (cidr & CID_CLASS_MASK) >> CID_CLASS_SHIFT;
 
-	/* ROM table */
-	if (cid_class == cidc_romtab) {
-		uint16_t designer = ((pidr >> 24) & 0xf00) | ((pidr >> 12) & 0x7f);
-		uint16_t partno = pidr & 0xfff;
-#if defined(ENABLE_DEBUG) && defined(PLATFORM_HAS_DEBUG)
-		/* Check SYSMEM bit */
-		uint32_t memtype = adiv5_mem_read32(ap, addr | ADIV5_ROM_MEMTYPE) &
-			ADIV5_ROM_MEMTYPE_SYSMEM;
+	const uint64_t pidr = adiv5_ap_read_pidr(ap, addr);
 
-		if (adiv5_dp_error(ap->dp)) {
-			DEBUG_WARN("Fault reading ROM table entry\n");
+	uint16_t designer;
+	if (pidr & PIDR_JEP106_USED) {
+		/* (OFFSET - 8) because we want it on bits 11:8 of new code, see "JEP-106 code list" */
+		designer = (pidr & PIDR_JEP106_CONT_MASK) >> (PIDR_JEP106_CONT_OFFSET - 8) |
+		           (pidr & PIDR_JEP106_CODE_MASK) >> PIDR_JEP106_CODE_OFFSET;
+
+		if (designer == JEP106_ERRATA_STM32WX || designer == JEP106_ERRATA_CS) {
```

Comment:
It might help to add a comment explaining why this is necessary so people aren't tempted to "fix" it or fiddle with it without understanding the full consequences of doing so.